docs/CMakeLists.txt

```diff
@@ -50,20 +48,17 @@ set(DOXYGEN_TOC_INCLUDE_HEADINGS 4)
 set(DOXYGEN_TYPEDEF_HIDES_STRUCT YES)
 set(DOXYGEN_UML_LIMIT_NUM_FIELDS 20)
 set(DOXYGEN_UML_LOOK YES)
-set(DOXYGEN_WARN_NO_PARAMDOC YES)
-set(DOXYGEN_WARN_IF_UNDOCUMENTED YES)
+set(DOXYGEN_WARN_IF_UNDOCUMENTED NO)
 set(DOXYGEN_WARN_IF_DOC_ERROR YES)
-set(DOXYGEN_WARN_AS_ERROR NO)
+set(DOXYGEN_WARN_IF_INCOMPLETE_DOC YES)
+set(DOXYGEN_WARN_NO_PARAMDOC YES)
+set(DOXYGEN_WARN_IF_UNDOC_ENUM_VAL YES)
+set(DOXYGEN_WARN_AS_ERROR FAIL_ON_WARNINGS)
 set(DOXYGEN_WARN_LOGFILE "doxygen.log")
 set(DOXYGEN_IMAGE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/images")
-
-FILE(GLOB_RECURSE HPP_SOURCES
-    "${DDC_SOURCE_DIR}/include/ddc/*.hpp")
-message(STATUS ${HPP_SOURCES})
-
 doxygen_add_docs(doc
         "${CMAKE_CURRENT_SOURCE_DIR}/About.md"
         "${CMAKE_CURRENT_SOURCE_DIR}/first_steps.md"
         "${CMAKE_CURRENT_SOURCE_DIR}/heat_equation.md"
-        ${HPP_SOURCES}
+        "${DDC_SOURCE_DIR}/include/ddc/"
         ALL)
```

Comment:
Was this file changed back intentionally?

Reply:
Yes indeed because I thought the recursive option of Doxygen would be enough.

Reply:
If it is sufficient then I agree that this is best, but I'm not sure that it is enough when you have `set(DOXYGEN_EXTRACT_ALL NO)`. It needs to be tested

Reply:
I have tested locally and there doesn't seem to be a big difference between these two methods. If anything your method retrieves more information

Reply:
But the errors definitely aren't triggered in this version

Reply:
If I make this change in my branch I still have the errors. Could the problem be due to the use of `set(DOXYGEN_EXTRACT_ALL NO)` which doesn't check the cache? Could this be overwriting your command-line options?